tt_metal/impl/dispatch/lock_free_queue.hpp

```diff
@@ -51,4 +51,25 @@ class LockFreeQueue {
         bool empty() const {
             return head.load() == tail.load();
         }
+        class Iterator : public std::iterator<std::forward_iterator_tag, T> {
+           private:
+            Node* current;
+
+           public:
+            Iterator(Node* start) : current(start) {}
+
+            Iterator& operator++() {
+                if (current != nullptr) {
+                    current = current->next;
+                }
+                return *this;
+            }
+
+            bool operator!=(const Iterator& other) const { return current != other.current; }
+
+            const T& operator*() const { return *(current->data); }
+        };
+
+        Iterator begin() { return Iterator(head.load()); }
+        Iterator end() { return Iterator(tail.load()); }
```

Comment:
Looks good!

Reply:
Glad someone appreciates it, I am a fan of using built in patterns and operators with overloaded functionality to support custom types.

You can now iterate on the lock free queue to do things like:

```
    for (const auto& cmd : queue) {
        cq.run_command_impl(cmd);
    }
```

:)